modules/core/src/test/java/org/apache/ignite/internal/managers/discovery/IgniteTopologyPrintFormatSelfTest.java

```diff
@@ -62,237 +64,209 @@
 
     /** {@inheritDoc} */
     @Override protected void beforeTest() throws Exception {
+        super.beforeTest();
+
         stopAllGrids();
     }
 
     /** {@inheritDoc} */
     @Override protected void afterTest() throws Exception {
         stopAllGrids();
 
-        if (log instanceof MockLogger)
-            ((MockLogger)log).clear();
+        super.afterTest();
     }
 
     /**
      * @throws Exception If failed.
      */
     @Test
     public void testServerLogs() throws Exception {
-        MockLogger log = new MockLogger();
-
-        log.setLevel(Level.INFO);
-
-        doServerLogTest(log);
+        doServerLogTest(false);
     }
 
     /**
      * @throws Exception If failed.
      */
     @Test
     public void testServerDebugLogs() throws Exception {
-        MockLogger log = new MockLogger();
-
-        log.setLevel(Level.DEBUG);
+        ((GridTestLog4jLogger)log).setLevel(Level.DEBUG);
 
-        doServerLogTest(log);
+        doServerLogTest(true);
     }
 
     /**
-     * @param log Logger.
+     * @param dbg Debug flag.
      * @throws Exception If failed.
      */
-    private void doServerLogTest(MockLogger log) throws Exception {
+    private void doServerLogTest(boolean dbg) throws Exception {
         String nodeId8;
 
+        testLog = new ListeningTestLogger(dbg, log);
+
+        LogListener aliveNodesLsnr = LogListener.matches(ALIVE_NODES_MSG).times(dbg ? 0 : 4).build();
```

Comment:
Is that enough to check that "aliveNodes=["  is present in logs, I believe that content also matters here.